src/components/forms/validator.ts

```diff
@@ -87,17 +89,9 @@ export const minMaxLength = (minLen: number, maxLen: number) => (value: string):
 
 export const ADDRESS_REPEATED_ERROR = 'Address already introduced'
 
-export const uniqueAddress = (addresses: string[] | List<string>): GenericValidatorType => (): ValidatorReturnType => {
-  // @ts-expect-error both list and array have signatures for map but TS thinks they're not compatible
-  const lowercaseAddresses = addresses.map((address) => address.toLowerCase())
-  const uniqueAddresses = new Set(lowercaseAddresses)
-  const lengthPropName = 'size' in addresses ? 'size' : 'length'
-
-  if (uniqueAddresses.size !== addresses?.[lengthPropName]) {
-    return ADDRESS_REPEATED_ERROR
-  }
-
-  return undefined
+export const uniqueAddress = (addresses: string[] | List<string> = []) => (address?: string): string | undefined => {
+  const addressExists = addresses.some(sameAddress.bind(null, address))
```

Comment:
I feel I have some readability issues here. I would do it in one more line but more explicit what is happening here.

```
const addressExists = addresses.some((addressFromList) => {
  return sameAddress(addressFromList, address)
})
```

What do you think?

Reply:
:laughing: Yes, sure.

To be honest, I came looking for this comment.

I must have been nostalgic when I did